src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -44,13 +46,19 @@ internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmen
         private TextFormattingRunProperties? _format;
         private readonly IClassificationType _hintClassification;
         private readonly ForegroundThreadAffinitizedObject _threadAffinitizedObject;
+        private readonly IThreadingContext _threadingContext;
+        private readonly IToolTipService _toolTipService;
+        private readonly Lazy<IStreamingFindUsagesPresenter> _streamingFindUsagesPresenter;
 
         public event EventHandler<SnapshotSpanEventArgs>? TagsChanged;
 
         public InlineParameterNameHintsTagger(InlineParameterNameHintsTaggerProvider taggerProvider, ITextView textView, ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
         {
+            _threadingContext = taggerProvider.ThreadingContext;
             _cache = new List<ITagSpan<IntraTextAdornmentTag>>();
             _threadAffinitizedObject = new ForegroundThreadAffinitizedObject(taggerProvider.ThreadingContext);
+            _toolTipService = taggerProvider.ToolTipService;
+            _streamingFindUsagesPresenter = taggerProvider.StreamingFindUsagesPresenter;
```

Comment:
You had exposed these as public readonly properties on the taggerProvider to avoid having to pass all the parameters in; does it make sense to also just pass the TaggerProvider to the tag object as well to avoid having to decompose it here?

Reply:
Yes, that would be much easier.